11/realnitinworks/module_generator.py

```diff
@@ -0,0 +1,76 @@
+"""
+Turn the following unix pipeline into Python code using generators
+$ for i in ../*/*py; do grep ^import $i|sed 's/import //g' ; done | sort | uniq -c | sort -nr
+   4 unittest
+   4 sys
+   3 re
+   3 csv
+   2 tweepy
+   2 random
+   2 os
+   2 json
+   2 itertools
+   1 time
+   1 datetime
+"""
+from collections import Counter
+from glob import glob
+
+
+PATH = "../../*/*.py"
+PATTERN = "import"
+
+
+def gen_files(path):
+    """
+    generate the names of all the files in the path
+
+    for i in ../*/*py
+    """
+    for file in glob(path):
+        yield file
+
+
+def gen_lines(files):
+    """
+    generate all lines from the interested files
+    """
+    for file in files:
+        with open(file, "r") as f:
+            yield from f
+
+
+def gen_grep(lines, pattern):
+    """
+    generate cleaned and filtered out lines by pattern
+
+    grep ^import $i|sed 's/import //g'
+    """
+    return (
+        line.replace(pattern, "").strip(f"\n ")  # Finally strip the newlines and spaces
```

Comment:
seems strip() is enough: 

```
>>> "hello\n ".strip()
'hello'
```

Reply:
nicely broken down over multiple lines

Reply:
> seems strip() is enough:
> 
> ```
> >>> "hello\n ".strip()
> 'hello'
> ```

Eye opener. All the while I was thinking it will strip only white spaces.